corehq/apps/users/models.py

```diff
@@ -3003,6 +3003,36 @@ class UserOptionToggles(models.Model):
     use_latest_build_cloudcare = models.BooleanField(default=False)
 
 
+def is_option_enabled(option, username=None, domain_name=None):
+    """
+    Returns whether an option is enabled for a user or their domain.
```

Comment:
It looks like users can override the domain settings - should domain settings take precedence?  That is, the current logic is:
```enabled_for_user or enabled_for_domain```
and I wonder whether it should be
```enabled_for_domain and enabled_for_user```
That is, if the domain setting is disabled, should users still have access to it?  I suppose that probably depends on the toggle.

And what about the UI representation?  If a user doesn't touch the setting, it'll fall back to the domain setting, so is this reflected in the UI?  When receiving an update from the user settings page, how can you tell whether they're setting a value for the setting deliberately versus leaving it at the default?  Maybe that commit will answer my questions.

Edit: it doesn't look like that commit addresses this point - one approach to this question would be to display the user setting only if it's enabled for the domain, and then in this function, switch the logic to
```enabled_for_domain and enabled_for_user```
to ensure that the domain setting always takes precedence

Reply:
This is meant to mimic a feature flag, where the feature is enabled if the toggle is enabled for the user or the domain.

I think what's really confusing here is the way I implemented this using two separate models, and then wrote this function that treats them the same in some ways, and different in others. My second swing at this change will have a function that is a lot more specific.